src/js/server/lib/i18n-node.js

```diff
@@ -0,0 +1,154 @@
+/*
+
+    Functions to assist in internationalising and localising content.
+
+*/
+/* eslint-env node */
+"use strict";
+var fluid = require("infusion");
+var gpii  = fluid.registerNamespace("gpii");
+
+var fs = require("fs");
+var path = require("path");
+var JSON5 = require("json5");
+
+require("./resolver");
+
+fluid.registerNamespace("gpii.handlebars.i18n");
+
+/**
+ *
+ * Derive a single message bundle from all available languages and locales, based on the `Accept-Language` headers
+ * passed as part of `request`.  Returns a combination of all messages found matching the following:
+ *
+ * 1. The user's locale.
+ * 2. The user's language.
+ * 3. The default locale.
+ *
+ * @param {Object} request - An express request object: http://expressjs.com/en/4x/api.html#req
+ * @param {Object} messageBundles - The full set of all message bundles available, keyed by locale or language.
+ * @param {String} defaultLocale - The default locale.
+ * @return {Object} - A map of messages for a single locale.
+ *
+ */
+gpii.handlebars.i18n.deriveMessageBundleFromRequest = function (request, messageBundles, defaultLocale) {
+    var header = fluid.get(request, "headers.accept-language");
+    return gpii.handlebars.i18n.deriveMessageBundleFromHeader(header, messageBundles, defaultLocale);
+};
+
+/**
+ *
+ * Load our all message bundles from one or more directories and organise them into a single object.
+ *
+ * @param {Array} messageDirs - An array of full or package-relative paths to directories containing message bundles.
+ * @param {String} defaultLocale - The default locale to use for files that do not provide language or locale data.
+ * @return {Object} - A combined bundle of message bundles, keyed by locales and languages.
+ *
+ */
+gpii.handlebars.i18n.loadMessageBundles = function (messageDirs, defaultLocale) {
+    defaultLocale = defaultLocale || "en_us";
+    var messageBundles = {};
+    var resolvedMessageDirs = gpii.express.hb.resolveAllPaths(messageDirs);
+
+    var filesByLocale   = {};
+    var filesByLanguage = {};
+
+    fluid.each(resolvedMessageDirs, function (messageDir) {
+        var resolvedPath = fluid.module.resolvePath(messageDir);
+        try {
+            var files = fs.readdirSync(resolvedPath);
+            fluid.each(files, function (filename) {
+                var fullPath = path.resolve(resolvedPath, filename);
+                var matchesLocale = filename.toLowerCase().match(/.+[-_]([a-z]{2}[-_][a-z]{2})\.json5?/);
+                var matchesLanguage = filename.toLowerCase().match(/.+[-_]([a-z]{2})\.json5?/);
+
+                if (matchesLocale) {
+                    var preferredLocale = matchesLocale[1];
+                    if (!filesByLocale[preferredLocale]) { filesByLocale[preferredLocale] = []; }
```

Comment:
Helpful core utility fluid.pushArray(filesByLocale, preferredLocale, fullPath) here - especially since the first branch is untested on line 76

Reply:
Replaced all three usages of the pattern with calls to `fluid.pushArray`.